File: DidYouKnow/app/src/main/java/com/example/didyouknow/services/api/TriviaService.java
package com.example.didyouknow.services.api;

import com.example.didyouknow.models.CategoryModel;
import com.example.didyouknow.models.InformationModel;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface TriviaService {
    @GET("/api/trivia/{categoryId}")
    Call<InformationModel> getInformation(@Path("categoryId") long categoryId);

    @GET("/api/trivia/categories")
    Call<CategoryModel[]> getCategories();

    @POST("/api/trivia")
    Call<InformationModel> postUserInformation(@Body() InformationModel model);
}

File: WebApp/DidYouKnowThat/src/app/services/trivia.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CategoryModel } from '../models/category.model';
import { HttpClient } from '@angular/common/http';
import { InformationModel } from '../models/information.model';

@Injectable({
    providedIn: 'root'
})
export class TriviaService {

    private baseUrl = 'https://cristiansarghe.com/api';

    constructor(
        private http: HttpClient
    ) { }

    public getCategories(): Observable<CategoryModel[]> {
        return this.http.get<CategoryModel[]>(`${this.baseUrl}/trivia/categories`);
    }

    public getUserAddedInformation(): Observable<InformationModel[]> {
        return this.http.get<InformationModel[]>(`${this.baseUrl}/moderatetrivia`);
    }

    public acceptUserAddedInformation(model: InformationModel): Observable<InformationModel> {
        return this.http.post<InformationModel>(`${this.baseUrl}/moderatetrivia`, model);
    }

    public deleteUserAddedInformation(id: number): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/moderatetrivia/${id}`);
    }

}